### PR TITLE
Remove attrs dependence from immutablecollections

### PR DESCRIPTION
**Pull request by berquist**
- [ ] Contains a change log entry for Towncrier (Should I do this even though the user interface hasn't changed?)

Closes https://github.com/isi-vista/immutablecollections/issues/37.

### CODE REVIEW

**Comment by berquist** [comment]
Oh, forgot to say that I didn't touch `ImmutableList`. Should I bother?

**Comment by rgabbard** [comment]
@berquist : don't bother with `ImmutableList`

File: immutablecollections/_immutablemultidict.py [hunk 1]
```diff
@@ -359,10 +368,13 @@ def _freeze_set_multidict(x: Mapping[KT, Iterable[VT]]) -> Mapping[KT, Immutable
...
 
-@attrs(frozen=True, slots=True, repr=False)
 class FrozenDictBackedImmutableSetMultiDict(ImmutableSetMultiDict[KT, VT]):
-    _dict: Mapping[KT, ImmutableSet[VT]] = attrib(converter=_freeze_set_multidict)
-    _len: Optional[int] = attrib(init=False, cmp=False, default=None)
+    __slots__ = "_dict", "_len"
+
+    # pylint:disable=assigning-non-slot,redefined-builtin
+    def __init__(self, dict: Mapping[KT, ImmutableSet[VT]], len: Optional[int] = None):
+        self._dict = _freeze_set_multidict(dict)
```
Comment (rgabbard) [inline immutablecollections/_immutablemultidict.py]: arg name here again

File: immutablecollections/_immutablemultidict.py [hunk 2]
```diff
@@ -359,10 +368,13 @@ def _freeze_set_multidict(x: Mapping[KT, Iterable[VT]]) -> Mapping[KT, Immutable
...
+        self._len = len
```
Comment (rgabbard) [inline immutablecollections/_immutablemultidict.py]: Not something you changed, but we should probably have a comment about why we cache `_len`

**Pull request merged**